File: backend/src/main/java/com/example/backend/note/Note.java
package com.example.backend.note;

import com.example.backend.notebook.Notebook;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Entity
@Table(name = "notes")
public class Note {
    @Id
    @SequenceGenerator(
            name = "note_sequence",
            sequenceName = "note_sequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "note_sequence"
    )

    @Column(name = "note_id", nullable = false)
    private Long id;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(nullable = false)
    private Notebook notebook;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private LocalDateTime date;

    public Note(Notebook notebook, String name, String content) {
        this.notebook = notebook;
        this.name = name;
        this.content = content;
        this.date = LocalDateTime.now();
    }
}

File: frontend/src/app/models/Note.ts
export class Note {
  id: number;
  name: string;
  content: string;
  date: Date;

  constructor(
    id: number | null,
    name: string | null,
    content: string | null,
    date: Date | null
  ) {
    if (id !== null) this.id = id;
    else this.id = 0;

    if (name !== null) this.name = name;
    else this.name = '';

    if (content !== null) this.content = content;
    else this.content = '';

    if (date !== null) this.date = date;
    else this.date = new Date();
  }
}
